builtin/providers/aws/resource_aws_config_delivery_channel.go

```diff
@@ -0,0 +1,171 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/configservice"
+)
+
+func resourceAwsConfigDeliveryChannel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsConfigDeliveryChannelPut,
+		Read:   resourceAwsConfigDeliveryChannelRead,
+		Update: resourceAwsConfigDeliveryChannelPut,
+		Delete: resourceAwsConfigDeliveryChannelDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "default",
+				ValidateFunc: validateMaxLength(256),
+			},
+			"s3_bucket_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"s3_key_prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"sns_topic_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+			"snapshot_delivery_properties": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"delivery_frequency": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateConfigExecutionFrequency,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsConfigDeliveryChannelPut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).configconn
+
+	name := d.Get("name").(string)
+	channel := configservice.DeliveryChannel{
+		Name:         aws.String(name),
+		S3BucketName: aws.String(d.Get("s3_bucket_name").(string)),
+	}
+
+	if v, ok := d.GetOk("s3_key_prefix"); ok {
+		channel.S3KeyPrefix = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("sns_topic_arn"); ok {
+		channel.SnsTopicARN = aws.String(v.(string))
+	}
+
+	if p, ok := d.GetOk("snapshot_delivery_properties"); ok {
+		propertiesBlocks := p.([]interface{})
+		block := propertiesBlocks[0].(map[string]interface{})
+
+		if v, ok := block["delivery_frequency"]; ok {
+			channel.ConfigSnapshotDeliveryProperties = &configservice.ConfigSnapshotDeliveryProperties{
+				DeliveryFrequency: aws.String(v.(string)),
+			}
+		}
+	}
+
+	input := configservice.PutDeliveryChannelInput{DeliveryChannel: &channel}
+
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
```

Comment:
As we only retry due to eventually consistent IAM I think it's more than sufficient.

Increasing the timeout means we'd also increase wait times for users that genuinely used incorrect IAM policies - i.e. you'd be waiting 5mins to realize that your policy is wrong or missing.